spark_expectations/sinks/utils/writer.py

```diff
@@ -83,20 +86,65 @@ def save_df_as_table(
             if config["options"] is not None and config["options"] != {}:
                 _df_writer = _df_writer.options(**config["options"])
 
-            if config["format"] == "bigquery":
-                _df_writer.option("table", table_name).save()
-            else:
-                _df_writer.saveAsTable(name=table_name)
-                self.spark.sql(
-                    f"ALTER TABLE {table_name} SET TBLPROPERTIES ('product_id' = '{self._context.product_id}')"
+            max_retries = 10
+            for attempt in range(1, max_retries + 2):
+                _log.info(
+                    "Writing records to table: %s, attempt: %s", table_name, attempt
                 )
-            _log.info("finished writing records to table: %s,", table_name)
-
+                try:
+                    if config["format"] == "bigquery":
+                        _df_writer.option("table", table_name).save()
+                    else:
+                        _df_writer.saveAsTable(name=table_name)
+                        if not stats_table:
```

Comment:
This logic can help with not having write conflicts, as stats table could be common across multiple jobs, while error and main table are unique in each job. Hoping this assumption is true, I think we can remove the below retries. 

Also may be good to add this in doc related to possible error scenarios if same error table is used across multiple jobs.

Reply:
I will remove the retries logic as only Alter table is the issue, I removed `alter` for stats table as it will have multiple products written into it. Error is unique to each table so it should not be a problem for error table.